### PR TITLE
add snax-alu kernel

### PR DESCRIPTION
**Pull request by jorendumoulin**
This PR copies the testcase for `snax-alu` from `snax_cluster` in this repository. 
I shall now embark on a journey to compile this example automatically from `linalg.generic` format.

### CODE REVIEW

File: kernels/streamer_alu/main.c [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    write_csr(0x3cc, MODE);
+    write_csr(0x3cd, LOOP_ITER);
+    write_csr(0x3ce, 1);
+
+    // Mark the end of the CSR setup cycles
+    uint32_t end_csr_setup = snrt_mcycle();
+
+    // Do this to poll the accelerator
+    while (read_csr(0x3cf)) {
+    };
```
Comment (JosseVanDelm) [inline kernels/streamer_alu/main.c:91]: In previous examples we put this in a baseline.c, I believe :thinking:

File: kernels/streamer_alu/Makefile [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Courtesy of Federico Ficarelli
+
+.DEFAULT_GOAL := all
+
+include ../../runtime/snax-alu.rules
+include ../../runtime/Makefile.rules
+
+TESTS =
+TESTS += streamer_mult.x
```
Comment (JosseVanDelm) [inline kernels/streamer_alu/Makefile:9]: wait, this works even though there are no files named streamer_mult.x?

File: kernels/streamer_alu/Makefile [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+CFLAGS += -std=gnu11
+CFLAGS += -Wall -Wextra
+
+%.acc_dialect.snax-opt.mlir: %.preprocfinal.mlir
+	$(SNAXOPT) $(SNAXOPTACCFLAGS) -o $@ $<
```
Comment (JosseVanDelm) [inline kernels/streamer_alu/Makefile:15]: Maybe remove this one, since it is not used?

**Review by JosseVanDelm — APPROVED** [review]
Looks good overall

File: kernels/streamer_alu/main.c (hunk 1 above)
Comment (JosseVanDelm) [reply kernels/streamer_alu/main.c:91]: Not sure actually, maybe just disregard

**Comment by JosseVanDelm** [comment]
> I shall now embark on a journey to compile this example automatically from linalg.generic format.

Good luck soldier

File: kernels/streamer_alu/Makefile (hunk 2 above)
Comment (jorendumoulin) [reply kernels/streamer_alu/Makefile:9]: yes through some makefile magic, streamer_mult.mlir is added in the next pr

**Pull request merged**